### DISCUSSION TITLE
[persistQueryClient v4] Persisted timestamp is updated each time the cache is updated

### DISCUSSION THREAD

**Discussion by TomPradat** [TanStack/query#6176] [General]
After doing some tests and looking at the code, I arrived at the conclusion that each time the cache is updated, the previously persisted timestamp is updated too.

Would it make sense to add this to the documentation? I feel like this isn't obvious and would benefit others.

To add a bit of context, I was wrongly expecting the `maxAge` to be the maximum amount of time a request would be persisted

**Comment by TkDodo** [comment]
it's the maximum amount of time the `cache` is persisted, since the cache is written as a whole. If you have 5 queries and refetch one of them, the maxAge for the whole cache re-sets.

if you want something different - v5 has a fine-grained persistence plugin to store each query separately with its own maxAge: https://tanstack.com/query/v5/docs/react/plugins/createPersister

**Reply by TomPradat** [reply]
Good to know, thanks for the answer 👍
